File: Chapter 03/Python/section_3_3_5_dip/after/forward.py
from .player import Player

"""
A Low-Level Detail.
* ARCHITECTURE NOTE: By inheriting from the Player ABC, these classes 
explicitly sign the contract. They are now "plug-and-play" components 
that the Coach can lead without needing to know their specific types.
"""
class Forward(Player):
    def perform_action(self):
        print("  [Action] Forward is attacking.")

File: Chapter 03/JS/section_3_3_5_dip/after/forward.js
/**
 * A Low-Level Detail.
 * * ARCHITECTURE NOTE: This class now fulfills a generic "Player" contract. 
 * By standardizing the method name to .performAction(), we allow this 
 * class to be swapped seamlessly with any other player type.
 */
class Forward {
    performAction() {
        console.log("  [Action] Forward is attacking.");
    }
}

module.exports = Forward;